modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java

```diff
@@ -1994,4 +2070,255 @@ private void processDelay(InetSocketAddress sockAddr) {
             reconnectionDelaysRemainder.put(sockAddr, delay);
         }
     }
+
+    /**
+     * JDBC implementation of {@link MarshallerContext}.
+     */
+    private class JdbcMarshallerContext extends BlockingJdbcChannel implements MarshallerContext {
+        /** Type ID -> class name map. */
+        private Map<Integer, String> cache = new ConcurrentHashMap<>();
+
+        /** {@inheritDoc} */
+        @Override public boolean registerClassName(
+            byte platformId,
+            int typeId,
+            String clsName,
+            boolean failIfUnregistered
+        ) throws IgniteCheckedException {
+
+            assert platformId == MarshallerPlatformIds.JAVA_ID
```

Comment:
don't we need this check on production, where java assertions are not enabled?

Reply:
First of all it helps to check that BinaryContext uses JAVA platform id when tries to register new type during test on test environment. If default configuration will change then we get the error. Besides there are 9 more assertions in this class (and not only here) that changes behavior of the method. So I suppose it is a common way to be sure that state is valid. If you know the best way to perform this check then I will be glad to hear.